### PR TITLE
[stable/unifi] unifi chart enhancements

### PR DESCRIPTION
**Pull request by billimek**
#### What this PR does / why we need it: 

* based on the persistent nature of this chart, using a default strategy Type of 'Recreate' instead of 'RollingUpdate' - with an optional override
* bumping unifi controller to the latest stable version (5.10.19)
* adding @mcronce to the OWNERS file
* updating label syntax to current helm standards (e.g.
`app.kubernetes.io/name`)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by billimek** [comment]
FYI @mcronce, adding you to the OWNERS file as part of this PR

**Comment by billimek** [comment]
Going to make some additional changes and re-submit

**Comment by billimek** [comment]
Re-opening after reviewing the [StatefulSet section of the helm chart guidelines](https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md#statefulset)

**Comment by mcronce** [comment]
@billimek Received 👍

**Comment by unguiculus** [comment]
Before we merge. Can you explain why it makes sense to use a StatefulSet here? Does Unifi support running with multiple replicas each having its on PV? Usually, this is applicable to databases that form a cluster and you'd need some logic to set up such a cluster which I don't see here.

**Comment by billimek** [comment]
Hi @unguiculus, 

The Unifi application embeds a mongoDB to persist information it needs behind-the-scenes.  Additionally, there is very little benefit to running this without some form of persistent storage given the nature of the workload (configuring and controlling devices).  

Given that a persistent volume is a 'must' to properly leverage this, and [based on the discussion here](https://github.com/helm/charts/issues/1863), it seems like a StatefulSet is more appropriate than a Deployment.

Also, when someone has this deployed with a storage backend like rook/ceph (or any type of storage that strictly forbids multiple pods accessing the same volume), upgrades of the chart will fail given that there will be more than one replica attempting to use the same volume due to the default deployment upgrade policy of RollingUpdate.

**Comment by unguiculus** [comment]
StatefulSets don't solve your problem. You can't generally say that anything that has state has to be a StatefulSet. As I mentioned, StatefulSets make sense for things like databases that can form a cluster. This doesn't seem to be the case here. If you run multiple replicas, it's like running multiple pods independently. They all have their own state/disk and don't know of each other. Imaging you have a load balancer in front of them, you'd probably access different replicas with every request which will cause anomalies because of different states.

**Comment by billimek** [comment]
Hi @unguiculus,

What are your thoughts about [this issue](https://github.com/helm/charts/issues/1863) and subsequent discussion?

This chart (as well as the ones covered by #12052 & #12049) are very much stateful applications, and require state to operate properly.  In particular, they should benefit from:

* Stable, persistent storage.
* Ordered, graceful deployment and scaling.
* Ordered, automated rolling updates.

The last two items are related to using a PVC with RWO (e.g. Block Storage): You cannot upgrade the chart because the upgrade cannot mount the storage, used by the old pod.

Isn't narrowing the definition of StatefulSet to be just clustered databases too narrow?  If you still do not believe or agree that StatefulSet is appropriate for these applications, would you allow these PRs to move forward if StatefulSet is introduced as an _option_ for the chart user to select in the values definition?

**Comment by unguiculus** [comment]
The issue you mentioned is wrong as such. It doesn't make sense to dismiss deployments out of hand as soon as state is involved. You need to look more carefully in order to decide whether a Deployment or a StatefulSet is the right thing to use. State itself on disk is problematic as soon as you want to scale things. There are no rolling updates for pods with PVs that require exclusive updates. StatefulSets don't solve this problem. Such pods must always be recreated. StatefulSets multiply state which doesn't help you and causes new problems if there is no coordination.

**Comment by unguiculus** [comment]
IMO, the bottom line is that apps that need a disk just don't support rolling updates. The should have `strategy.type: Recreate` and no more than one replica.

File: stable/unifi/templates/ingress.yaml
Comment (davidkarlsen) [inline stable/unifi/templates/ingress.yaml]: Ingress is still beta only, isn't it? https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#ingress-v1beta1-extensions

**Review by davidkarlsen — CHANGES_REQUESTED** [review]

**Comment by davidkarlsen** [comment]
It very much depends on the workload how to deal with disk (and indeed recreate + scale = 1 is one way to tackle this) - however I think that the discussion in https://github.com/helm/charts/issues/1863 and docs in https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/ points out that statefulsets are favourable due to the stable claim id's and offer a better chance of data surviving - this is important or workloads when the data is "valuable" - e.g. not any random scratch-data that can be lost.

**Comment by scottrigby** [comment]
@davidkarlsen Good point. 

@billimek re the discussion you mentioned, I reopened and repurposed that conversation for the purpose of documenting rules of thumb around this. See https://github.com/helm/charts/issues/1863#issuecomment-476324748

File: stable/unifi/templates/ingress.yaml
Comment (billimek) [reply stable/unifi/templates/ingress.yaml]: Very nice catch, thanks!  I'll fix this.

**Comment by billimek** [comment]
Hi friends, thank you so much for the really great discussion on this topic!

@unguiculus your amazing attention to detail is honestly really appreciated - thank you!

@davidkarlsen, @scottrigby, thanks for the additional feedback as well - this clarifies some misconceptions I had about when to use statefulsets.

Based on this discussion I'll update this PR (and the other two similar charts) to:

* Use Deployment instead of StatefulSet as is the current mode
* Default the upgrade strategy to be `Recreate` instead of `RollingUpdate` with the option to change this via a values parameter
* Bump minor version instead of major given that it won't be a breaking change any more

**Comment by billimek** [comment]
@unguiculus & @davidkarlsen these changes are made, FYI!

File: stable/unifi/OWNERS
Comment (unguiculus) [inline stable/unifi/OWNERS:3]: OWNERS must also be listed as maintainers in `Chart.yaml`. Should we invite @mcronce as collaborator? He'll be expected to actively contribute to the chart, review PRs, triage issues, etc.

**Review by unguiculus — CHANGES_REQUESTED** [review]
The number of replicas should not longer be configurable.

File: stable/unifi/OWNERS
Comment (billimek) [reply stable/unifi/OWNERS:3]: @mcronce is now added as a maintainer in the `Chart.yml`, and it would be great if they could be added as a collaborator as well, thank you!

Comment (unguiculus) [reply stable/unifi/OWNERS:3]: done

**Review by unguiculus — APPROVED** [review]

**Pull request merged**
